Make respond_to available for API-Only applications

Matthijsy opened the pull request:
### Summary
This PR fixes that the `respond_to` was not found in the API only mode. I also found a little bug that when removing an authorized application in API only mode the response code was 200 with an empty payload instead of a 204. 

Resolved #1494

nbulaj reviewed (approved):
LGTM :+1: 

Let's finish instructions from the bot (ignore about tests) and it's ready to merge

Matthijsy commented:
Thanks @nbulaj ! I have updated the changelog and squashed everything into one commit. I cannot find any tests related to the api-only mode, so I did not include/update any tests. Should I still do so or is it fine like this?
